### PR TITLE
Remove Prettier 2 support and change to pure ESM

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

Prettier 3 supports the ESM format, so we can now provide this package as pure ESM.
See https://prettier.io/blog/2023/07/05/3.0.0.html

---

Also, this PR updates all outdated dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,16 +9,17 @@
...
-    ".prettierrc.js"
+    "prettier.config.js"
   ],
   "scripts": {
     "prepare": "husky install",
     "test": "npm run lint",
     "format": "prettier . --write",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:formatting": "prettier . --check --ignore-path .gitignore",
+    "lint:formatting": "prettier . --check",
```
Comment (ybiquitous) [inline package.json:22]: [note] Prettier 3 ignores `.gitignore` by default.
See https://prettier.io/blog/2023/07/05/3.0.0.html#cli

**Comment by ybiquitous** [comment]
https://github.com/stylelint/eslint-config-stylelint/pull/226 tests this PR.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

(Incidentally, [Prettier 3.0.0](https://prettier.io/blog/2023/07/05/3.0.0.html) is a good read as we approach our migration to ESM.)

**Comment by ybiquitous** [comment]
Yeah, Prettier 3 ships both CJS and ESM. 😃 
https://github.com/prettier/prettier/blob/072c819ca3800c7943f87574e6ad089af2b5fb96/package.json#L15-L22

**Comment by ybiquitous** [comment]
I'm going to release `@stylelint/prettier-config` 3.0.0 soon.

**Comment by ybiquitous** [comment]
We need one more PR #23 for a new release 😅

**Pull request merged**